tests/Spec.hs

```diff
@@ -1877,7 +1894,278 @@ main = hspec $ do
+          , BuildStatusChanged (Sha "2b2") "default" Project.BuildSucceeded
+          , PullRequestCommitChanged (PullRequestId 1) (Sha "2b2")
+          ]
+        results = defaultResults {resultIntegrate = [Right (Sha "1b2"), Right (Sha "2b2"), Right (Sha "3b2")]
+                                 , resultPush = [PushOk, PushRejected "", PushOk, PushOk]}
+        actions = snd $ runActionCustom results $ handleEventsTest events state
+      actions `shouldBe`
+        [ AIsReviewer "deckard"
+        , ALeaveComment (PullRequestId 1)
+                        "<!-- Hoff: ignore -->\nPull request approved for merge by @deckard, rebasing now."
+        , ATryIntegrate "Merge #1: Improvements...\n\n\
+                        \Approved-by: deckard\n\
+                        \Auto-deploy: false\n"
+                        (PullRequestId 1, Branch "refs/pull/1/head", Sha "ab1")
+                        []
+                        False
+        , ALeaveComment (PullRequestId 1) "<!-- Hoff: ignore -->\nRebased as 1b2, waiting for CI …"
+        , ALeaveComment (PullRequestId 1) "<!-- Hoff: ignore -->\n[CI job :yellow_circle:](example.com/1b2) started."
+        , ATryForcePush (Branch "fst") (Sha "1b2")
+        , ATryPromote (Sha "1b2")
+        , ATryIntegrate "Merge #1: Improvements...\n\n\
+                        \Approved-by: deckard\n\
+                        \Auto-deploy: false\n"
+                        (PullRequestId 1, Branch "refs/pull/1/head", Sha "ab1")
+                        []
+                        False
+        , ALeaveComment (PullRequestId 1) "<!-- Hoff: ignore -->\nRebased as 2b2, waiting for CI …"
+        , ALeaveComment (PullRequestId 1) "<!-- Hoff: ignore -->\n[CI job :yellow_circle:](example.com/2b2) started."
+        , ATryForcePush (Branch "fst") (Sha "2b2")
+        , ATryPromote (Sha "2b2")
+        , ACleanupTestBranch (PullRequestId 1)
+        ]
+
+    it "correctly waits for a previous PR to be promoted" $ do
+      let
+        state
+          = Project.insertPullRequest (PullRequestId 1)
+              (Branch "fst") masterBranch (Sha "ab1") "Improvements..." (Username "huey")
+          $ Project.insertPullRequest (PullRequestId 2)
+              (Branch "snd") masterBranch (Sha "ab2") "... Of the ..." (Username "dewey")
+          $ Project.emptyProjectState
+        events =
+          [ CommentAdded (PullRequestId 1) "deckard" "@bot merge"
+          , BuildStatusChanged (Sha "1b2") "default" (Project.BuildStarted "example.com/1b2")
+          , CommentAdded (PullRequestId 2) "deckard" "@bot merge"
+          , BuildStatusChanged (Sha "2b2") "default" (Project.BuildStarted "example.com/2b2")
+          , BuildStatusChanged (Sha "2b2") "default" Project.BuildSucceeded
+          , BuildStatusChanged (Sha "1b2") "default" Project.BuildSucceeded
+          , PullRequestCommitChanged (PullRequestId 1) (Sha "1b2")
+          , PullRequestCommitChanged (PullRequestId 2) (Sha "2b2")
```

Comment:
So the pull request commit changed event does the push. What happens if you
reverse the order of these two? Does `2b2` (which contains the changes in `1b2`) get
merged into master first before `1b2`?

Reply:
So only a single PR will ever be 'awaiting promotion'?

Reply:
Only a single PR per project, yes.